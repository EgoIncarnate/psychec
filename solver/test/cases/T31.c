void f()
{
    /*
        We treat void* as a top type. Below:
            - T1 can be void* or a pointer to an orphan type.
            - T2 can be void or an orphan type.
        When a named type stands for a void*, entirely, we make it as such. Otherwise,
        the pointer-declarator appears in the declaration, we make a pointer to orphan.
     */
    T1 a = malloc(4);
    T2* b = malloc(4);
}
