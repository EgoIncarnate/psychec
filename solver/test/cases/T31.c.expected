typedef int /*<<< orphan or scalar*/ T2;
typedef void* T1;
